tests/botserver_mock.py

```diff
@@ -0,0 +1,170 @@
+        """Run the command as the specified user in the test environment."""
+        testmsg = [{'type': 'message', 'user': exec_user, 'text': msg, 'client_msg_id': '738e4beb-d50e-42a4-a60e-3fafd4bd71da',
+                    'team': 'UNITTESTTEAMID', 'channel': 'UNITTESTCHANNELID', 'event_ts': '1549715670.002000', 'ts': '1549715670.002000'}]
+        self.exec_message(testmsg)
+
+    def test_reaction(self, reaction, exec_user="normal_user"):
+        """Run the specified reaction as the specified user in the test environment."""
+        testmsg = [{'type': 'reaction_added', 'user': exec_user, 'item': {'type': 'message', 'channel': 'UNITTESTCHANNELID', 'ts': '1549117537.000500'},
+                    'reaction': reaction, 'item_user': 'UNITTESTUSERID', 'event_ts': '1549715822.000800', 'ts': '1549715822.000800'}]
+
+        self.exec_message(testmsg)
+
+    def check_for_response_available(self):
+        return len(self.slack_wrapper.message_list) > 0
+
+    def check_for_response(self, expected_result):
+        """ Check if the simulated slack responses contain an expected result. """
+        for msg in self.slack_wrapper.message_list:
+            if expected_result in msg.message:
+                return True
+
+        return False
+
+    def exec_message(self, testmsg):
+        self.running = True
+
+        while self.running:
+            try:
+                self.load_config()
+                self.slack_wrapper = SlackWrapperMock("testapikey")
+
+                if self.slack_wrapper.connected:
+                    self.load_bot_data()
+                    READ_WEBSOCKET_DELAY = 1  # 1 second delay between reading from firehose
+
+                    # Might even pass the bot server for handlers?
+                    handler_factory.initialize(self.slack_wrapper, self.bot_id, self)
+
+                    # Main loop
+                    while self.running:
+                        message = testmsg
+                        if message:
+                            reaction, channel, ts, reaction_user = self.parse_slack_reaction(message)
+
+                            if reaction:
+                                log.debug("Received reaction : {} ({})".format(reaction, channel))
+                                handler_factory.process_reaction(
+                                    self.slack_wrapper, reaction, ts, channel, reaction_user)
+
+                            command, channel, ts, user = self.parse_slack_message(message)
+
+                            if command:
+                                log.debug("Received bot command : {} ({})".format(command, channel))
+                                handler_factory.process(self.slack_wrapper, self,
+                                                        command, ts, channel, user)
+
+                            # We're in test context, immediately stop
+                            self.running = False
+                else:
```

Comment:
Similar to my other comment about inverted logic, if we do `if !self.slack_wrapper.connected` for this case, and then `break` (if we can) we can un-indend the whole connected block